File: tests/fixtures/typescript_project/src/pricing.ts
/**
 * Pricing module — fixture for tests.
 */

export function calculatePrice(itemId: number, qty: number): number {
  const base = getBasePrice(itemId);
  return base * qty;
}

export function applyDiscount(price: number, pct: number): number {
  return price * (1 - pct / 100);
}

function getBasePrice(itemId: number): number {
  const prices: Record<number, number> = { 1: 10, 2: 20, 3: 5 };
  return prices[itemId] ?? 0;
}

File: tests/fixtures/python_project/modules/pricing.py
"""Pricing module — fixture for tests."""


def get_base_price(item_id: int) -> float:
    """Return base price for an item."""
    prices = {1: 10.0, 2: 20.0, 3: 5.0}
    return prices.get(item_id, 0.0)


def apply_discount(price: float, pct: float) -> float:
    """Apply a percentage discount."""
    return price * (1 - pct / 100)


def calculate_price(item_id: int, qty: int) -> float:
    """Calculate final price with all applicable discounts."""
    base = get_base_price(item_id)
    if qty > 100:
        base = apply_discount(base, 10)
    return base * qty
